cli_tools/gce_vm_image_import/importer/processor_test.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/stretchr/testify/assert"
+	v1 "google.golang.org/api/compute/v1"
+)
+
+func TestProcessorProvider_InspectDataDisk(t *testing.T) {
+	processorProvider := defaultProcessorProvider{
+		ImportArguments{
+			Inspect:     true,
+			WorkflowDir: "testdata",
+			DataDisk:    true,
+		},
+		mockComputeDiskClient{},
+		mockDiskInspector{true},
+	}
+
+	processors, err := processorProvider.provide(persistentDisk{})
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(processors), "there should be 1 processor, got %v", len(processors))
+	_, ok := processors[0].(*dataDiskProcessor)
+	assert.True(t, ok, "processor is not dataDiskProcessor")
+}
+
+func TestProcessorProvider_InspectUEFI(t *testing.T) {
+	tests := []struct {
+		isUEFIDisk               bool
+		isInputArgUEFICompatible bool
+	}{
+		{isUEFIDisk: true, isInputArgUEFICompatible: false},
+		{isUEFIDisk: false, isInputArgUEFICompatible: false},
+		{isUEFIDisk: true, isInputArgUEFICompatible: true},
+		{isUEFIDisk: false, isInputArgUEFICompatible: true},
+	}
+	for i, tt := range tests {
+		name := fmt.Sprintf("%v. inspect disk: disk is UEFI: %v, input arg UEFI compatible: %v", i+1, tt.isUEFIDisk, tt.isInputArgUEFICompatible)
+		t.Run(name, func(t *testing.T) {
+			processorProvider := defaultProcessorProvider{
+				ImportArguments{
+					Inspect:        true,
+					WorkflowDir:    "testdata",
+					OS:             "ubuntu-1804",
+					UefiCompatible: tt.isInputArgUEFICompatible,
+				},
+				mockComputeDiskClient{},
+				mockDiskInspector{tt.isUEFIDisk},
+			}
+
+			pd := persistentDisk{uri: "old-uri"}
+			processors, err := processorProvider.provide(pd)
+			assert.NoError(t, err)
+			assert.Equal(t, 3, len(processors), "there should be 3 processors, got %v", len(processors))
+			diskInspectionProcessor, ok := processors[0].(*diskInspectionProcessor)
+			assert.True(t, ok, "the 1st processor is not diskInspectionDiskProcessor")
+			diskMutationProcessor, ok := processors[1].(*diskMutationProcessor)
+			assert.True(t, ok, "the 2nd processor is not diskMutationProcessor")
+			bootableDiskProcessor, ok := processors[2].(*bootableDiskProcessor)
+			assert.True(t, ok, "the 3rd processor is not bootableDiskProcessor")
+
+			pd, err = diskInspectionProcessor.process(pd)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.isUEFIDisk, pd.isUEFIDetected)
+			assert.Equal(t, tt.isInputArgUEFICompatible || tt.isUEFIDisk, pd.isUEFICompatible)
+
+			pd, err = diskMutationProcessor.process(pd)
+			assert.NoError(t, err)
+			if tt.isUEFIDisk && !tt.isInputArgUEFICompatible {
+				assert.Truef(t, strings.HasSuffix(pd.uri, "uefi"), "UEFI Disk URI should have suffix 'uefi', actual: %v", pd.uri)
+			} else {
+				assert.Falsef(t, strings.HasSuffix(pd.uri, "uefi"), "Disk URI shouldn't have suffix 'uefi', actual: %v", pd.uri)
+			}
+
+			pd, err = bootableDiskProcessor.process(pd)
+			assert.NotEmpty(t, bootableDiskProcessor.workflow.Vars["source_disk"].Value)
```

Comment:
This is a bit of an E2E test as all 3 types of processors are partially tested here. This test func should only care if processor provider works correctly. The test sfor each of the 3 processors should be separated.

Reply:
split to 4: 1 for provider, 3 for 3 different processors